source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,141 @@
+
+            List<AdditionalTempData.PlayerInfo> pinfolist = new List<AdditionalTempData.PlayerInfo>();
+
+            foreach (GameData.PlayerInfo player in players)
+            {
+                AdditionalTempData.PlayerInfo pinfo = new AdditionalTempData.PlayerInfo();
+                byte pid = player.PlayerId;
+                pinfo.PlayerName = player.PlayerName;
+                pinfo.CurrentRole = Role.GetRole(player);
+                pinfo.Faction = pinfo.CurrentRole.Faction;
+                pinfo.Modifier = Modifier.GetModifier(player);
+                pinfo.TasksCompleted = 0;
+                pinfo.TasksTotal = 0;
+
+                foreach (var task in player.Tasks)
+                {
+                    if (task.Complete == true)
+                    {
+                        pinfo.TasksCompleted++;
+                    }
+                    pinfo.TasksTotal++;
+                }
+                pinfolist.Add(pinfo);
+            }
+            AdditionalTempData.playerData = pinfolist;
+        }
+
+        public static void LoadGameSummary(EndGameManager __instance)
+        {
+            var position = Camera.main.ViewportToWorldPoint(new Vector3(0f, 1f, Camera.main.nearClipPlane));
+            GameObject roleSummary = UnityEngine.Object.Instantiate(__instance.WinText.gameObject);
+            roleSummary.transform.position = new Vector3(__instance.ExitButton.transform.position.x + 0.1f, position.y - 0.1f, -14f);
+            roleSummary.transform.localScale = new Vector3(1f, 1f, 1f);
+
+            var roleSummaryText = new StringBuilder();
+
+            roleSummaryText.AppendLine("End-of-game Summary:");
+
+            foreach (var data in AdditionalTempData.playerData)
+            {
+
+                //this way works, dont know why the other doesnt tbh.
+                var won = "   ";
+                foreach (WinningPlayerData winner in TempData.winners)
+                {
+                    Reactor.PluginSingleton<TownOfUs>.Instance.Log.LogDebug($"WinnerName=#{winner.Name}# vs PlayerName=#{data.PlayerName}#");
+                    if (winner.Name == data.PlayerName)
+                    {
+                        Reactor.PluginSingleton<TownOfUs>.Instance.Log.LogDebug($"TRUE");
+                        won = $"[<color=#FAD934FF>W</color>]";
+                    }
+                }
+
+                //this doesnt work for some reason, it stops both lovers from showing as winners.
+                /*var won = "";
+                foreach (WinningPlayerData winner in TempData.winners)
+                {
+                    Reactor.PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"WinnerName=#{winner.Name}# vs PlayerName=#{data.PlayerName}#");
+                    if (winner.Name == data.PlayerName)
+                    {
+                        Reactor.PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"TRUE");
+                        won = $"[<color=#FAD934FF>W</color>]";
+                    }
```

Comment:
I believe the issue was you forgetting to add `break;` causing the loop to continue (and assigns `won` as the blank white spaces).

Reply:
yes, apparantly i forgot a break after the won, this is why i put it as just if() without an else before, as now it re-assigns blank spaces until it finds the right player, i have changed this though, thanks for pointing this out.